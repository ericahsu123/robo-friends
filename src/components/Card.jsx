import React, { useState } from 'react';

const Card = (props) => {
  const [likesCount, setLikesCount] = useState(0);

  const onIncrementLikes = () => {
    setLikesCount((prevLikesCount) => {
      if (prevLikesCount < 5) {
        return prevLikesCount + 1;
      }
      return prevLikesCount;
    });
  };
  return (
    <div className="ui centered card">
      <div className="image">
        <img
          src={`https://robohash.org/${props.name}.png?bgset=bg1 `}
          alt="robot"
          style={{ backgroundColor: 'rgb(185, 226, 224)' }}
        />
      </div>
      <div className="content">
        <div className="center aligned header">{props.name}</div>
        <div className="center aligned description">{props.email}</div>
        <span className="right floated">
          <i
            className="star outline like icon"
            onClick={onIncrementLikes}
            data-testid="add-count"
          ></i>
          <span data-testid="likes-count">{likesCount}</span>
          <span> </span>
          <span>stars</span>
        </span>
      </div>
    </div>
  );
};

export default Card;
