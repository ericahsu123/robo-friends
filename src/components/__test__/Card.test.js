import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Card from '../Card';

afterEach(cleanup);

// test('<Heading/> matches snapshot', () => {
//     const tree = renderer.create(<Heading />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

describe('Card component matches snapshot', () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Card component loads correctly', () => {
  const { container } = render(<Card />);
  test('renders correctly', () => {
    expect(container).toHaveTextContent(/stars/i);
  });
});

describe('State is managed correctly', () => {
  it('likes count starts at 0', () => {
    const { getByTestId } = render(<Card />);
    const count = getByTestId('likes-count');
    expect(count.innerHTML).toBe('0');
  });

  it('should add 1 to count', () => {
    const { getByTestId } = render(<Card />);
    const count = getByTestId('likes-count');
    const add = getByTestId('add-count');
    // Click 1 time
    for (let i = 0; i < 1; i++) {
      fireEvent.click(add);
    }
    expect(count.innerHTML).toBe('1');
  });

  it('should add 2 to count', () => {
    const { getByTestId } = render(<Card />);
    const count = getByTestId('likes-count');
    const add = getByTestId('add-count');
    // Click 2 times
    for (let i = 0; i < 2; i++) {
      fireEvent.click(add);
    }
    expect(count.innerHTML).toBe('2');
  });

  it('should add 3 to count', () => {
    const { getByTestId } = render(<Card />);
    const count = getByTestId(`likes-count`);
    const add = getByTestId(`add-count`);
    // Click 3 times
    for (let i = 0; i < 3; i++) {
      fireEvent.click(add);
    }
    expect(count.innerHTML).toBe('3');
  });

  it('should add 4 to count', () => {
    const { getByTestId } = render(<Card />);
    const count = getByTestId(`likes-count`);
    const add = getByTestId(`add-count`);
    // Click 4 times
    for (let i = 0; i < 4; i++) {
      fireEvent.click(add);
    }
    expect(count.innerHTML).toBe('4');
  });

  it('should add 5 to count', () => {
    const { getByTestId } = render(<Card />);
    const count = getByTestId(`likes-count`);
    const add = getByTestId(`add-count`);
    // Click 5 times
    for (let i = 0; i < 5; i++) {
      fireEvent.click(add);
    }
    expect(count.innerHTML).toBe('5');
  });

  it('should add up no more than 5', () => {
    const { getByTestId } = render(<Card />);
    const count = getByTestId(`likes-count`);
    const add = getByTestId(`add-count`);
    for (let i = 0; i < 10; i++) {
      fireEvent.click(add);
    }
    expect(count.innerHTML).toBe('5');
  });
});
