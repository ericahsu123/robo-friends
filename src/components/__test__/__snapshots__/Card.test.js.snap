// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="ui centered card"
>
  <div
    className="image"
  >
    <img
      alt="robot"
      src="https://robohash.org/undefined.png?bgset=bg1 "
      style={
        Object {
          "backgroundColor": "rgb(185, 226, 224)",
        }
      }
    />
  </div>
  <div
    className="content"
  >
    <div
      className="center aligned header"
    />
    <div
      className="center aligned description"
    />
    <span
      className="right floated"
    >
      <i
        className="star outline like icon"
        data-testid="add-count"
        onClick={[Function]}
      />
      <span
        data-testid="likes-count"
      >
        0
      </span>
      <span>
         
      </span>
      <span>
        stars
      </span>
    </span>
  </div>
</div>
`;
